src/granite/transactions.cr

```diff
@@ -66,14 +66,14 @@ module Granite::Transactions
       end
       begin
         {% if primary_type.id == "Int32" %}
-          @{{primary_name}} = @@adapter.insert(@@table_name, fields, params, lastval: true).to_i32
+          @{{primary_name}} = @@adapter.insert(@@table_name, fields, params, lastval: "{{primary_name}}").to_i32
         {% elsif primary_type.id == "Int64" %}
-          @{{primary_name}} = @@adapter.insert(@@table_name, fields, params, lastval: true)
+          @{{primary_name}} = @@adapter.insert(@@table_name, fields, params, lastval: "{{primary_name}}")
         {% elsif primary_auto == true %}
           {% raise "Failed to define #{@type.name}#save: Primary key must be Int(32|64), or set `auto: false` for natural keys.\n\n  primary #{primary_name} : #{primary_type}, auto: false\n" %}
         {% else %}
           if @{{primary_name}}
-            @@adapter.insert(@@table_name, fields, params, lastval: false)
+            @@adapter.insert(@@table_name, fields, params, lastval: nil)
```

Comment:
why is this changed from false to nil?

Reply:
It looks like `lastval` is being changed from a `Bool` to a `String?` to pass the name of the primary key.  Maybe we should rename that field?

Reply:
@drujensen that's exactly it. If there's another name you'd rather I use then I would be happy to update it.